### PR TITLE
Lynn/create matching schema

### PR DESCRIPTION
**Pull request by lynnlow175**

### CODE REVIEW

**Comment by Daviancold** [comment]
Do note this satisfies #90 and not #89

File: backend/matching-service/src/models/matching.model.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        required: true,
+    },
+    matchToken: {
+        type: String,
+        required: true,
+        unique: true,
+    },
+    createdAt: {
+        type: Date,
+        default: Date.now,
```
Comment (Daviancold) [inline backend/matching-service/src/models/matching.model.ts:27]: Good use of unix epoch time

**Review by Daviancold — APPROVED** [review]
LGTM

**Review by glemenneo — CHANGES_REQUESTED** [review]
Should create some indexes to support more advanced queries. Based on the existing data table design, seems like need to support sorting by complexity and date. Should add an index for that, can refer to question schema for a reference. 

Can also refer to the index section of the mongoose guide for instructions on how to create indexes.
https://mongoosejs.com/docs/guide.html

**Review by glemenneo — APPROVED** [review]

**Pull request merged**